### ISSUE TITLE
Typing a message is slow

### ISSUE THREAD

**Issue by ralphtheninja** [deltachat/deltachat-desktop#802]
- Platform (linux/mac/windows): linux
- Delta Chat Desktop Version: latest master
- Expected behavior: typing/editing a message should be smooth
- Actual behavior: it's slow
- Steps to reproduce the problem: type quickly or hold down key
- Screenshots:
- Logs:

Typing a message quickly or holding down a key (maybe erasing something using backspace) is slow. The log spits out `DC_EVENT_MSGS_CHANGED` for each key pressed:

```
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
main/deltachat DEBUG [] Core Event DC_EVENT_MSGS_CHANGED
```

Maybe we can do some timeout in desktop so we don't call core on each keypress, but we do some backoff (or whatever it's called), say 50 ms and when that kicks in we call core to update the message being typed (I'm guessing we're updating a draft message).

**Comment by Jikstra** [comment]
I guess this event is getting triggered every time we do a setDraft call (which is currently getting debounced, or at least that's how i implemented it). DC_EVENT_MSGS_CHANGED probably does a rerender every time of everything. This should get solved during the chatlist refactoring.

**Comment by Simon-Laux** [comment]
should be solved with #837 (setDraft isn't called as much now)
reopen if it is still there @ralphtheninja
